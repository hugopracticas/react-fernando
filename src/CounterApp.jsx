import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
  };

  const handleLess = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <div>
      <h1>CounterApp</h1>
      <h2 data-testid="test-value">{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button data-testid="test-button-less" onClick={handleLess}>
        -1
      </button>
      <button aria-label="btn-reset" onClick={handleReset}>
        RESET
      </button>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  value: 1590,
};

export default CounterApp;
