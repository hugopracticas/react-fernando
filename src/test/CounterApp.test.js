import { screen, fireEvent, render } from "@testing-library/react";
import CounterApp from "../CounterApp";

describe("Test in CounterApp", () => {
  const value = 100;
  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<CounterApp value={value} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar el valor inicial de 100", () => {
    const { getByTestId } = render(<CounterApp value={value} />);
    expect(getByTestId("test-value").innerHTML).toBe("100");
  });

  test("Debe de incrementar con el boton +1", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("+1"));
    expect(screen.getByText("101")).toBeTruthy();
  });

  test("Debe de decrementar con el boton -1", () => {
    const { getByTestId } = render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("-1"));
    //screen.debug();
    expect(screen.getByText("99")).toBeTruthy();
  });

  test("Debe de funcionar el boton de reset", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    //fireEvent.click(screen.getByText("RESET"));
    fireEvent.click(screen.getByRole("button", { name: "btn-reset" }));
    expect(screen.getByText(value)).toBeTruthy;
  });
});
