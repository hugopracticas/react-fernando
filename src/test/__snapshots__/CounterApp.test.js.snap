// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in CounterApp Debe de hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1>
      CounterApp
    </h1>
    <h2
      data-testid="test-value"
    >
      100
    </h2>
    <button>
      +1
    </button>
    <button
      data-testid="test-button-less"
    >
      -1
    </button>
    <button
      aria-label="btn-reset"
    >
      RESET
    </button>
  </div>
</div>
`;
